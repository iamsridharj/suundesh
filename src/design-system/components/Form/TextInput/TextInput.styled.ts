import { forDesktop } from '@design-system/breakpoints'
import styled from 'styled-components'

export const StyledTextInput = styled.input<{
  $transparentBorder: boolean
}>`
  font-size: 18px;
  margin: 0 8px;
  width: 100%;
  border-radius: 4px;
  padding: 7px 18px;
  border: 1px solid ${({ theme, $transparentBorder }) => {
    return $transparentBorder ? "transparent" : theme.colors.raven.raven60;
  }};
  border-color: ${({ theme, $transparentBorder }) => {
      return $transparentBorder ? "transparent" : theme.colors.raven.raven60;
    }};
  outline: none;

  &:focus {
    border: 1px solid ${({ $transparentBorder }) => {
      return $transparentBorder ? "transparent" : "solid";
    }};
    border-color: ${({ theme, $transparentBorder }) => {
      
      return $transparentBorder ? "transparent" : theme.colors.raven.raven60;
    }};
    outline: none;
  }

  ${
    forDesktop`
      width: 95%;
    `
  }
`
