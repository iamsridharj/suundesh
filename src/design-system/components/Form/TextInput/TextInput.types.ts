export interface TextInputProps
  extends Omit<React.InputHTMLAttributes<HTMLInputElement>, 'onChange'> {
  value: string
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
  ariaLabel?: string
  transparentBorder?: boolean;
  isError?: boolean;
}
