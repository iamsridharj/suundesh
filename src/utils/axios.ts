import axios, { AxiosRequestConfig, AxiosInstance } from 'axios';

const baseURL = "https://dev-api.konfhub.com";

const axiosWrapper: AxiosInstance = axios.create({
    baseURL: baseURL,
    headers: {
        'Content-Type': 'application/json',
    },
});

interface BackendResponse<T> {
  status: string;
  data: T;
}

const get = async <T>(url: string, params: Record<string, any> = {}, config: AxiosRequestConfig = {}): Promise<BackendResponse<T>> => {
  try {
    const response = await axiosWrapper.get<BackendResponse<T>>(url, { params, ...config });
    return response.data;
  } catch (error) {
    throw error;
  }
};

const post = async <T>(url: string, data: any = {}, config: AxiosRequestConfig = {}): Promise<BackendResponse<T>> => {
  try {
    const response = await axiosWrapper.post<BackendResponse<T>>(url, data, { ...config });
    return response.data;
  } catch (error) {
    throw error;
  }
};

export { axiosWrapper, get, post };
