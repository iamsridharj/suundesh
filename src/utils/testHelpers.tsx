// src/utils/testHelpers.tsx
import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { useTheme } from "src/hooks/ui/useTheme";

const RenderWithTheme = ({ children }: { children: React.ReactNode }) => {
  const theme = useTheme();

  return (
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  );
};

const render = (ui: React.ReactElement, options = {}) => {
  return rtlRender(
    <RenderWithTheme>
      {ui}
    </RenderWithTheme>,
    options
  );
};

export * from '@testing-library/react';
export { render };
