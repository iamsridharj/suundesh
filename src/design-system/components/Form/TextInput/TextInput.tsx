import React from "react";
import { StyledTextInput } from "./TextInput.styled"
import { TextInputProps } from "./TextInput.types"

export const TextInput: React.FC<TextInputProps> = ({ type, value, ariaLabel, onChange, transparentBorder, ...props }) => {

    return (
        <StyledTextInput
            type={type}
            value={value}
            aria-label={ariaLabel || 'text input'}
            onChange={(e) => onChange(e)}
            $transparentBorder={!!transparentBorder}
            {...props}
        />
    )
}