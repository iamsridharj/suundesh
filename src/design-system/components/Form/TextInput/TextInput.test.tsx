import React from 'react';
import { render, fireEvent } from 'src/utils/testHelpers';
import { TextInput } from './TextInput';

describe('TextInput Unit Tests', () => {
    it('renders with default aria-label', () => {
        const { getByLabelText } = render(
            <TextInput type="text" value="test" onChange={() => { }} />
        );
        expect(getByLabelText('text input')).toBeInTheDocument();
    });

    it('renders with provided aria-label', () => {
        const { getByLabelText } = render(
            <TextInput type="text" value="test" ariaLabel="custom label" onChange={() => { }} />
        );
        expect(getByLabelText('custom label')).toBeInTheDocument();
    });

    it('calls onChange handler on input change', () => {
        const handleChange = jest.fn();
        const { getByLabelText } = render(
            <TextInput type="text" value="test" onChange={handleChange} />
        );

        fireEvent.change(getByLabelText('text input'), { target: { value: 'new value' } });
        expect(handleChange).toHaveBeenCalledTimes(1);
    });

    it('renders with transparent border when transparentBorder is true', () => {
        const { getByLabelText } = render(
            <TextInput type="text" value="test" onChange={() => { }} transparentBorder />
        );
        expect(getByLabelText('text input')).toHaveStyle('border: 1px solid transparent');
    });

    it('renders with additional props', () => {
        const { getByPlaceholderText } = render(
            <TextInput
                type="text"
                value="test"
                onChange={() => { }}
                placeholder="Enter text here"
            />
        );
        expect(getByPlaceholderText('Enter text here')).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(
            <TextInput
                type="text"
                value="test"
                onChange={() => { }}
                placeholder="Enter text here"
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
