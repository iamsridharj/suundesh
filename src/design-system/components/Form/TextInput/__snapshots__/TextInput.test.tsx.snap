// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Unit Tests matches snapshot 1`] = `
<DocumentFragment>
  <input
    aria-label="text input"
    class="sc-beySPh cuIecB"
    placeholder="Enter text here"
    type="text"
    value="test"
  />
</DocumentFragment>
`;
